server/data/data-jmap/src/test/java/org/apache/james/jmap/api/preview/MessagePreviewStoreContract.java

```diff
@@ -114,6 +120,47 @@ default void storeShouldBeIdempotent() {
             .isEqualTo(PREVIEW_1);
     }
 
+    @Test
+    default void concurrentStoreShouldOverrideOldValueWhenSameMessageId() throws Exception {
+        int threadCount = 10;
+        int stepCount = 100;
+
+        ConcurrentTestRunner.builder()
+            .operation((thread, step) -> Mono.from(testee()
```

Comment:
that would make sense for ConcurrentTestRunner to support Mono as parameter of `operation`